### DISCUSSION TITLE
Define Material Property as Boundary Condition

### DISCUSSION THREAD

**Discussion by FanZhangchen** [idaholab/moose#24106] [Q&A General] [answered]
Hi MOOSE Team,

The material property I need to implement in my problem is the slip rate and which is space-dependent, and the boundary condition of my problem is also related to the slip rate. So I have a question about whether the material property could be defined as the boundary condition in MOOSE? Or in other ways, could I get the nodal coordinate to use Boolean to define the boundary conditions inside the material property module?

Best,

ZC

**Comment by hugary1995** [comment, marked as answer]
It depends. 

If you want to apply a nodal (Dirichlet) BC, there is no straightforward way of accessing material properties from the node. But you can always provide a spatially-varying function in a `FunctionDirichletBC`.

If you want to apply an integrated (Neumann, Robin, etc.) BC, yes you can define the BC as a function of some material properties. See this example: https://github.com/idaholab/moose/discussions/24108#discussioncomment-5648628

**Reply by hugary1995** [reply]
I said

> If you want to apply a nodal (Dirichlet) BC, there is no straightforward way of accessing material properties from the node.

So it is technically possible to "make" material properties nodal. You'll essentially need an additional layer of projection. Let us know if you want to pursue that option.

**Reply by FanZhangchen** [reply]
Could you give more information about the additional layer of projection?

**Reply by GiudGiud** [reply]
@hugary1995

**Reply by iomags2211** [reply]
@hugary1995 can you give more information on this or has it been answered elsewhere?

**Reply by hugary1995** [reply]
You need to extrapolate the material property from quadrature points to nodes. You can do either a least squares fit or an L2 projection.

For least squares fit, you can follow what's been done in the nodal patch recovery system.

For L2 projection, you need a lower dimensional block on the boundary, and then add a MaterialPropertyValue kernel to perform the projection.
